File: tools/cocos批量设置分包/batch-set-bundle.js
#!/usr/bin/env node

const path = require('path');
const fs = require('fs');

/**
 * 批量设置Bundle配置脚本
 * 使用方法: node batch-set-bundle.js
 */
class BatchBundleSetter {
    constructor() {
        this.subpackagesPath = path.join(__dirname, 'assets/subpackages');
        this.successCount = 0;
        this.errorCount = 0;
        this.errors = [];
    }

    async execute() {
        console.log('🚀 开始批量设置Bundle配置...');
        console.log(`📁 分包路径: ${this.subpackagesPath}`);
        
        try {
            const subpackages = await this.getSubpackages();
            console.log(`📦 找到 ${subpackages.length} 个分包`);
            
            for (let i = 0; i < subpackages.length; i++) {
                const subpackage = subpackages[i];
                console.log(`\n[${i + 1}/${subpackages.length}] 处理分包: ${subpackage}`);
                
                try {
                    await this.setBundleForSubpackage(subpackage);
                    this.successCount++;
                    console.log(`✅ ${subpackage} 设置成功`);
                } catch (error) {
                    this.errorCount++;
                    this.errors.push({ subpackage, error: error.message });
                    console.log(`❌ ${subpackage} 设置失败: ${error.message}`);
                }
                
                // 添加延迟避免过快操作
                await this.sleep(50);
            }
            
            this.printSummary();
            
        } catch (error) {
            console.error('❌ 批量设置失败:', error.message);
        }
    }

    async getSubpackages() {
        return new Promise((resolve, reject) => {
            fs.readdir(this.subpackagesPath, (err, files) => {
                if (err) {
                    reject(new Error(`读取分包目录失败: ${err.message}`));
                    return;
                }
                
                const directories = files.filter(file => {
                    const fullPath = path.join(this.subpackagesPath, file);
                    try {
                        return fs.statSync(fullPath).isDirectory();
                    } catch {
                        return false;
                    }
                });
                
                resolve(directories);
            });
        });
    }

    async setBundleForSubpackage(subpackageName) {
        const subpackagePath = path.join(this.subpackagesPath, subpackageName);
        const metaPath = path.join(this.subpackagesPath, `${subpackageName}.meta`);
        
        return new Promise((resolve, reject) => {
            if (!fs.existsSync(subpackagePath)) {
                reject(new Error(`分包目录不存在: ${subpackagePath}`));
                return;
            }

            if (!fs.existsSync(metaPath)) {
                reject(new Error(`未找到.meta文件: ${metaPath}`));
                return;
            }

            try {
                const metaContent = fs.readFileSync(metaPath, 'utf8');
                const meta = JSON.parse(metaContent);

                // 只添加honor-minigame属性，保持其他配置不变
                if (!meta.compressionType) {
                    meta.compressionType = {};
                }
                if (!meta.compressionType['honor-minigame']) {
                    meta.compressionType['honor-minigame'] = 'zip';
                }
                
                if (!meta.optimizeHotUpdate) {
                    meta.optimizeHotUpdate = {};
                }
                if (!meta.optimizeHotUpdate['honor-minigame']) {
                    meta.optimizeHotUpdate['honor-minigame'] = false;
                }
                
                if (!meta.inlineSpriteFrames) {
                    meta.inlineSpriteFrames = {};
                }
                if (!meta.inlineSpriteFrames['honor-minigame']) {
                    meta.inlineSpriteFrames['honor-minigame'] = false;
                }
                
                if (!meta.isRemoteBundle) {
                    meta.isRemoteBundle = {};
                }
                if (!meta.isRemoteBundle['honor-minigame']) {
                    meta.isRemoteBundle['honor-minigame'] = true;
                }

                const updatedMetaContent = JSON.stringify(meta, null, 2);
                fs.writeFileSync(metaPath, updatedMetaContent, 'utf8');

                resolve();
            } catch (error) {
                reject(new Error(`处理.meta文件失败: ${error.message}`));
            }
        });
    }

    printSummary() {
        console.log('\n' + '='.repeat(50));
        console.log('📊 批量设置Bundle配置完成');
        console.log('='.repeat(50));
        console.log(`✅ 成功: ${this.successCount} 个分包`);
        console.log(`❌ 失败: ${this.errorCount} 个分包`);
        
        if (this.errors.length > 0) {
            console.log('\n❌ 失败详情:');
            this.errors.forEach(({ subpackage, error }) => {
                console.log(`  - ${subpackage}: ${error}`);
            });
        }
        
        console.log('\n🎉 批量设置完成！');
    }

    sleep(ms) {
        return new Promise(resolve => setTimeout(resolve, ms));
    }
}

// 执行脚本
const batchSetter = new BatchBundleSetter();
batchSetter.execute().catch(console.error);

File: tools/cocos批量设置分包/batch-set-bundle.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import os
import json
import time
from pathlib import Path

class BatchBundleSetter:
    def __init__(self):
        self.subpackages_path = Path(__file__).parent / "assets" / "subpackages"
        self.success_count = 0
        self.error_count = 0
        self.errors = []

    def execute(self):
        print("🚀 开始批量设置Bundle配置...")
        print(f"📁 分包路径: {self.subpackages_path}")
        
        try:
            subpackages = self.get_subpackages()
            print(f"📦 找到 {len(subpackages)} 个分包")
            
            for i, subpackage in enumerate(subpackages):
                print(f"\n[{i + 1}/{len(subpackages)}] 处理分包: {subpackage}")
                
                try:
                    self.set_bundle_for_subpackage(subpackage)
                    self.success_count += 1
                    print(f"✅ {subpackage} 设置成功")
                except Exception as error:
                    self.error_count += 1
                    self.errors.append({"subpackage": subpackage, "error": str(error)})
                    print(f"❌ {subpackage} 设置失败: {error}")
                
                # 添加延迟避免过快操作
                time.sleep(0.05)
            
            self.print_summary()
            
        except Exception as error:
            print(f"❌ 批量设置失败: {error}")

    def get_subpackages(self):
        """获取所有分包目录"""
        if not self.subpackages_path.exists():
            raise Exception(f"分包目录不存在: {self.subpackages_path}")
        
        directories = []
        for item in self.subpackages_path.iterdir():
            if item.is_dir():
                directories.append(item.name)
        
        return sorted(directories)

    def set_bundle_for_subpackage(self, subpackage_name):
        """为单个分包设置Bundle配置"""
        subpackage_path = self.subpackages_path / subpackage_name
        meta_path = self.subpackages_path / f"{subpackage_name}.meta"
        
        if not subpackage_path.exists():
            raise Exception(f"分包目录不存在: {subpackage_path}")
        
        if not meta_path.exists():
            raise Exception(f"未找到.meta文件: {meta_path}")
        
        try:
            # 读取.meta文件
            with open(meta_path, 'r', encoding='utf-8') as f:
                meta = json.load(f)
            
            # 只添加honor-minigame属性，保持其他配置不变
            if "compressionType" not in meta:
                meta["compressionType"] = {}
            if "honor-minigame" not in meta["compressionType"]:
                meta["compressionType"]["honor-minigame"] = "zip"
            
            if "optimizeHotUpdate" not in meta:
                meta["optimizeHotUpdate"] = {}
            if "honor-minigame" not in meta["optimizeHotUpdate"]:
                meta["optimizeHotUpdate"]["honor-minigame"] = False
            
            if "inlineSpriteFrames" not in meta:
                meta["inlineSpriteFrames"] = {}
            if "honor-minigame" not in meta["inlineSpriteFrames"]:
                meta["inlineSpriteFrames"]["honor-minigame"] = False
            
            if "isRemoteBundle" not in meta:
                meta["isRemoteBundle"] = {}
            if "honor-minigame" not in meta["isRemoteBundle"]:
                meta["isRemoteBundle"]["honor-minigame"] = True
            
            # 保存.meta文件
            with open(meta_path, 'w', encoding='utf-8') as f:
                json.dump(meta, f, indent=2, ensure_ascii=False)
            
        except Exception as error:
            raise Exception(f"处理.meta文件失败: {error}")

    def print_summary(self):
        """打印结果统计"""
        print("\n" + "=" * 50)
        print("📊 批量设置Bundle配置完成")
        print("=" * 50)
        print(f"✅ 成功: {self.success_count} 个分包")
        print(f"❌ 失败: {self.error_count} 个分包")
        
        if self.errors:
            print("\n❌ 失败详情:")
            for error_info in self.errors:
                print(f"  - {error_info['subpackage']}: {error_info['error']}")
        
        print("\n🎉 批量设置完成！")

if __name__ == "__main__":
    batch_setter = BatchBundleSetter()
    batch_setter.execute()
